### ISSUE TITLE
The `sd_imageURLForState:` method returns incorrect result.

### ISSUE THREAD

**Issue by iwill** [rs/SDWebImage#2001]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 10.3.3
 SDWebImage Version      | 4.1.0
 Integration Method      | cocoapods
 Xcode Version           | Xcode 8.3.3
 Repro rate              | all the time (100%)
 Repro with our demo prj | no
 Demo project link       | no

### Issue Description and Steps

In the UIButton+WebCache.m, the [line 69](https://github.com/rs/SDWebImage/blob/master/SDWebImage/UIButton%2BWebCache.m#L69) and [line 115](https://github.com/rs/SDWebImage/blob/master/SDWebImage/UIButton%2BWebCache.m#L115) are the same -- `imageURLStorage` for image and backgroundImage are the same.

```objc
self.imageURLStorage[@(state)] = url;
```

Which means the following statements will get incorrect result.

```objc
NSURL *imgURL = [NSURL URLWithString:@"https://www.img-url.com/"], *bgURL = [NSURL URLWithString:@"https://www.bg-url.com/"];

UIControlState theSameState = UIControlStateNormal;

UIButton *button = [UIButton new];
[button sd_setImageWithURL:imgURL forState:theSameState];
[button sd_setBackgroundImageWithURL:bgURL forState:theSameState];

NSAssert([[button sd_imageURLForState:theSameState] isEqual:imgURL], @"the return value should be %@", imgURL);
```

**Comment by dreampiggy** [comment]
Reproduced. Seems an Bug.

**Comment by bpoplauschi** [comment]
Fixed by #2002
